### ISSUE TITLE
VSTest Publish test results links tests with build but not release

### ISSUE THREAD

**Issue by vinnyrose** [microsoft/azure-pipelines-tasks#10504]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: VsTest

list here (V# not needed):  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

## Environment
- Server - Azure Pipelines or TFS on-premises?
    
    - If using TFS on-premises, provide the version: 
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: can provide on request

- Agent Version: 2.129.0
- Agent OS: Microsoft Windows Server 2016 Datacenter

## Issue Description

Using a VSTest task on a deployment group, the test results are being attached to the triggering build instead of on the release on which the task is ran on.

Configuration of the tasks:

*VSTest Platform installer Yaml*
```
steps:
- task: VisualStudioTestPlatformInstaller@1
  displayName: 'Visual Studio Test Platform Installer'
  inputs:
    versionSelector: latestStable
```

*VSTest Yaml*
```
steps:
- task: VSTest@2
  displayName: 'Test run for Test plans'
  inputs:
    testAssemblyVer2: |
     **/*....dll
     !**\obj\**
    searchFolder: '$(System.DefaultWorkingDirectory)/Tests'
    testFiltercriteria: 'TestCategory=AppA'
    vsTestVersion: toolsInstaller
    runInParallel: true
    batchingBasedOnAgentsOption: customBatchSize
    customBatchSizeValue: 10000
    testRunTitle: 'Test Run Name'
    diagnosticsEnabled: false
    rerunFailedTests: true
  continueOnError: true
```

### Task logs

[tasklog.zip](https://github.com/microsoft/azure-pipelines-tasks/files/3228318/tasklog.zip)

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

### Error logs

none

**Comment by ShreyasRmsft** [comment]
@vinnyrose what version of tfs are you on?

**Comment by vinnyrose** [comment]
@ShreyasRmsft Hi Shreyas, we are on azure devops services (VSTS).

**Comment by ShreyasRmsft** [comment]
@vinnyrose terribly sorry for the delay in response. Yes this is a know bug and we will be fixing this soon. I will update here once the fix is in. Thanks

**Comment by dgee2** [comment]
I'm seeing the same behaviour on Azure Devops Server 2019.0.1.
Is there a fix or a known workaround for this issue or is there any information around when this is likely to be fixed?

**Comment by ShreyasRmsft** [comment]
We have made the fix and will be rolling it out in the next hosted release (approx in 3-4 weeks).

**Comment by ShreyasRmsft** [comment]
This has been fixed in task version 2.159.x and rollout has already started and should complete in a few days. Please re-open the ticket if the issue persists even after getting the task with version 2.159.x.
